pkg/models/openpitrix/applicationversions.go

```diff
@@ -428,7 +428,8 @@ func (c *applicationOperator) DoAppVersionAction(versionId string, request *Acti
 			return err
 		}
 
-		if !reflect.DeepEqual(&app.Spec, &appInStore.Spec) {
+		if !reflect.DeepEqual(&app.Spec, &appInStore.Spec) || !reflect.DeepEqual(app.Labels[constants.CategoryIdLabelKey], appInStore.Labels[constants.CategoryIdLabelKey]) {
+			appCopy.Labels[constants.CategoryIdLabelKey] = app.Labels[constants.CategoryIdLabelKey]
 			appCopy := appInStore.DeepCopy()
```

Comment:
```
appCopy := appInStore.DeepCopy()
appCopy.Labels[constants.CategoryIdLabelKey] = app.Labels[constants.CategoryIdLabelKey]
```

Reply:
The appCopy variable is declared in the following line, so the verification fails. You need to adjust the code position.

Reply:
Sorry, the position of the code was added to the wrong position, we have made corrections and resubmitted the code